pkg/odo/cli/component/ui/ui.go

```diff
@@ -0,0 +1,228 @@
+package ui
+
+import (
+	"fmt"
+	"github.com/golang/glog"
+	"github.com/redhat-developer/odo/pkg/catalog"
+	"github.com/redhat-developer/odo/pkg/component"
+	"github.com/redhat-developer/odo/pkg/occlient"
+	"github.com/redhat-developer/odo/pkg/odo/cli/ui"
+	"github.com/redhat-developer/odo/pkg/odo/genericclioptions"
+	"github.com/redhat-developer/odo/pkg/odo/util/validation"
+	"github.com/redhat-developer/odo/pkg/util"
+	"gopkg.in/AlecAivazis/survey.v1"
+	"sort"
+)
+
+// SelectComponentType lets the user to select the builder image (name only) in the prompt
+func SelectComponentType(options []catalog.CatalogImage) string {
+	var componentType string
+	prompt := &survey.Select{
+		Message: "Which component type would wish to create",
+		Options: getComponentTypeNameCandidates(options),
+	}
+	err := survey.AskOne(prompt, &componentType, survey.Required)
+	ui.HandleError(err)
+	return componentType
+}
+
+func getComponentTypeNameCandidates(options []catalog.CatalogImage) []string {
+	result := make([]string, len(options))
+	for i, option := range options {
+		result[i] = option.Name
+	}
+	sort.Strings(result)
+	return result
+}
+
+// SelectImageTagInteractively lets the user to select a specific tag for the previously selected builder image in a prompt
+func SelectImageTagInteractively(options []catalog.CatalogImage, selectedComponentType string) string {
```

Comment:
I'd remove the `Interactively` part for this function since you haven't used it in the other functions.